### ISSUE TITLE
Installing Karaf features completely breaks Opencast

### ISSUE THREAD

**Issue by JulianKniephoff** [opencast/opencast#5151]
Trying to `feature:install` any feature, or chucking it into `deploy`---even if it is empty!---currently completely brings Opencast down.

Spending way too much time on finding out why I found that this is not actually an Opencast problem. I can reproduce it in a fresh copy of Karaf. It still concerns us, of course, so I'm making this issue to track it.

I will link the appropriate upstream issue here as soon as I'm able to create it. But for those who can't wait: It's basically because Karaf wants to refresh the `javax.persistence` bundle because different parts of the system don't agree on the wiring it's supposed to get. Basically every bundle of ours (transitively) depends on it so everything is restarted which Opencast can't handle. (I guess that part **is** an Opencast problem, but it's not what this issue is concerned with AFAIAC.

**To Reproduce**

`feature:install basically-whatever-you-want` or chuck a `features.xml` thingy into `deploy`.

**Expected behavior**

The feature is installed and I can use it and can continue to use Opencast undisturbed.

**Stack trace**

The log shows Opencast basically shutting down and restarting, but it doesn't quite make it; it will continue to spam the log with a wide range of exceptions, too many to list here.

**Server environment:**
 - OS: tested on latest macOS and Arch Linux, but likely doesn't matter.
 - Distribution: dev/allinone, but it shouldn't matter.
- Installation method: from source, but see above
 - Version: tested with `develop` but I also tested Karaf versions as far back as 4.1, so everything that uses that will likely be affected as well.

**Additional context**

🤷

**Comment by mliradelc** [comment]
Related to #4610 

What shows you if you do `feature:install -v <what-you-want>`?

**Comment by JulianKniephoff** [comment]
I saw #4610. Maybe it's a special case of this issue? Didn't really look too closely into it, tbh. 🤷‍♀️

For an empty feature I get this:

```
karaf@root()> feature:install -t -v foo
Adding features: foo/[1.0.0,1.0.0]
No deployment change.
  Bundles to refresh:
    com.mysema.querydsl.jpa/3.7.4 (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    javax.persistence/2.2.0.v201708071007 (Should be wired to: org.apache.aries.jpa.container/2.7.3 (through [javax.persistence/2.2.0.v201708071007] osgi.wiring.package; filter:="(&(osgi.wiring.package=org.osgi.service.jpa)(version>=1.1.0))"))
    opencast-admin-ui/15.0.0.SNAPSHOT (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    opencast-adopter-registration-impl/15.0.0.SNAPSHOT (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    opencast-analyze-mediapackage-workflowoperation/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-animate-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-animate-impl/15.0.0.SNAPSHOT (Wired to opencast-animate-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-animate-workflowoperation/15.0.0.SNAPSHOT (Wired to opencast-animate-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-asset-manager-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-asset-manager-impl/15.0.0.SNAPSHOT (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    opencast-asset-manager-static-file-authorization/15.0.0.SNAPSHOT (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    opencast-asset-manager-storage-aws/15.0.0.SNAPSHOT (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    opencast-asset-manager-storage-fs/15.0.0.SNAPSHOT (Wired to opencast-asset-manager-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-asset-manager-workflowoperation/15.0.0.SNAPSHOT (Wired to opencast-asset-manager-impl/15.0.0.SNAPSHOT which is being refreshed)
    opencast-authorization-manager/15.0.0.SNAPSHOT (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    opencast-authorization-xacml/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-caption-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-caption-impl/15.0.0.SNAPSHOT (Wired to opencast-caption-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-capture-admin-service-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-capture-admin-service-impl/15.0.0.SNAPSHOT (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    opencast-comments-workflowoperation/15.0.0.SNAPSHOT (Wired to opencast-event-comment/15.0.0.SNAPSHOT which is being refreshed)
    opencast-common/15.0.0.SNAPSHOT (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    opencast-common-jpa-impl/15.0.0.SNAPSHOT (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    opencast-composer-ffmpeg/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-composer-service-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-composer-workflowoperation/15.0.0.SNAPSHOT (Wired to opencast-composer-service-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-conductor/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-cover-image-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-cover-image-impl/15.0.0.SNAPSHOT (Wired to opencast-cover-image-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-cover-image-workflowoperation/15.0.0.SNAPSHOT (Wired to opencast-cover-image-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-crop-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-crop-ffmpeg/15.0.0.SNAPSHOT (Wired to opencast-crop-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-crop-workflowoperation/15.0.0.SNAPSHOT (Wired to opencast-crop-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-dictionary-api/15.0.0.SNAPSHOT (Wired to opencast-mpeg7/15.0.0.SNAPSHOT which is being refreshed)
    opencast-dictionary-regexp/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-distribution-service-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-distribution-service-aws-s3/15.0.0.SNAPSHOT (Wired to opencast-distribution-service-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-distribution-service-aws-s3-api/15.0.0.SNAPSHOT (Wired to opencast-distribution-service-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-distribution-service-download/15.0.0.SNAPSHOT (Wired to opencast-distribution-service-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-distribution-service-streaming-wowza/15.0.0.SNAPSHOT (Wired to opencast-distribution-service-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-distribution-workflowoperation/15.0.0.SNAPSHOT (Wired to opencast-distribution-service-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-dublincore/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-editor-service/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-editor-service-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-elasticsearch-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-elasticsearch-impl/15.0.0.SNAPSHOT (Wired to opencast-elasticsearch-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-elasticsearch-index/15.0.0.SNAPSHOT (Wired to opencast-elasticsearch-impl/15.0.0.SNAPSHOT which is being refreshed)
    opencast-email-template-service-api/15.0.0.SNAPSHOT (Wired to opencast-workflow-service-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-email-template-service-impl/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-engage-ui/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-event-comment/15.0.0.SNAPSHOT (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    opencast-execute-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-execute-impl/15.0.0.SNAPSHOT (Wired to opencast-execute-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-execute-workflowoperation/15.0.0.SNAPSHOT (Wired to opencast-execute-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-external-api/15.0.0.SNAPSHOT (Wired to opencast-dublincore/15.0.0.SNAPSHOT which is being refreshed)
    opencast-fileupload/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-incident-workflowoperation/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-index-service/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-ingest-download-service-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-ingest-download-service-impl/15.0.0.SNAPSHOT (Wired to opencast-ingest-download-service-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-ingest-download-service-workflowoperation/15.0.0.SNAPSHOT (Wired to opencast-ingest-download-service-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-ingest-service-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-ingest-service-impl/15.0.0.SNAPSHOT (Wired to opencast-ingest-service-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-inspection-service-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-inspection-service-ffmpeg/15.0.0.SNAPSHOT (Wired to opencast-inspection-service-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-inspection-workflowoperation/15.0.0.SNAPSHOT (Wired to opencast-inspection-service-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-kernel/15.0.0.SNAPSHOT (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    opencast-list-providers-service/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-live-schedule-api/15.0.0.SNAPSHOT (Wired to opencast-dublincore/15.0.0.SNAPSHOT which is being refreshed)
    opencast-live-schedule-impl/15.0.0.SNAPSHOT (Wired to opencast-distribution-service-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-logging-workflowoperation/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-lti/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-lti-service-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-lti-service-impl/15.0.0.SNAPSHOT (Wired to opencast-dublincore/15.0.0.SNAPSHOT which is being refreshed)
    opencast-mattermost-notification-workflowoperation/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-message-broker-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-metadata/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-metadata-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-metadata-to-acl-workflowoperation/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-metrics-exporter/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-mpeg7/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-notification-workflowoperation/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-oaipmh/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-oaipmh-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-oaipmh-persistence/15.0.0.SNAPSHOT (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    opencast-presets/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-publication-service-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-publication-service-configurable/15.0.0.SNAPSHOT (Wired to opencast-distribution-service-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-publication-service-oaipmh/15.0.0.SNAPSHOT (Wired to opencast-distribution-service-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-publication-service-youtube-v3/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-redirect/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-rename-files-workflowoperation/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-runtime-info/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-scheduler-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-scheduler-impl/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-schema/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-search-service-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-search-service-feeds/15.0.0.SNAPSHOT (Wired to opencast-search-service-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-search-service-impl/15.0.0.SNAPSHOT (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    opencast-security-aai/15.0.0.SNAPSHOT (Wired to opencast-kernel/15.0.0.SNAPSHOT which is being refreshed)
    opencast-security-ldap/15.0.0.SNAPSHOT (Wired to opencast-kernel/15.0.0.SNAPSHOT which is being refreshed)
    opencast-security-lti/15.0.0.SNAPSHOT (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    opencast-security-shibboleth/15.0.0.SNAPSHOT (Wired to opencast-kernel/15.0.0.SNAPSHOT which is being refreshed)
    opencast-series-service-api/15.0.0.SNAPSHOT (Wired to opencast-dublincore/15.0.0.SNAPSHOT which is being refreshed)
    opencast-series-service-impl/15.0.0.SNAPSHOT (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    opencast-serviceregistry/15.0.0.SNAPSHOT (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    opencast-shared-filesystem-utils/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-silencedetection-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-silencedetection-impl/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-smil-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-smil-impl/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-smil-workflowoperation/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-solr/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-sox-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-sox-impl/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-sox-workflowoperation/15.0.0.SNAPSHOT (Wired to opencast-composer-service-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-speech-to-text-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-speech-to-text-impl/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-speech-to-text-workflowoperation/15.0.0.SNAPSHOT (Wired to opencast-inspection-service-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-static/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-static-file-service-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-static-file-service-impl/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-statistics-export-service-api/15.0.0.SNAPSHOT (Wired to opencast-elasticsearch-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-statistics-export-service-impl/15.0.0.SNAPSHOT (Wired to opencast-elasticsearch-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-statistics-provider-influx/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-statistics-provider-random/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-statistics-service-impl/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-statistics-writer-workflowoperation/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-termination-state-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-termination-state-impl/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-textanalyzer-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-textanalyzer-impl/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-textanalyzer-workflowoperation/15.0.0.SNAPSHOT (Wired to opencast-composer-service-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-textextractor-tesseract/15.0.0.SNAPSHOT (Wired to opencast-textanalyzer-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-themes/15.0.0.SNAPSHOT (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    opencast-themes-workflowoperation/15.0.0.SNAPSHOT (Wired to opencast-composer-service-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-timelinepreviews-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-timelinepreviews-ffmpeg/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-timelinepreviews-workflowoperation/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-tobira/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-urlsigning-service-impl/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-user-interface-configuration/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-userdirectory/15.0.0.SNAPSHOT (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    opencast-userdirectory-ldap/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-videoeditor-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-videoeditor-ffmpeg-impl/15.0.0.SNAPSHOT (Wired to opencast-inspection-service-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-videoeditor-workflowoperation/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-videogrid-service-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-videogrid-service-impl/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-videogrid-workflowoperation/15.0.0.SNAPSHOT (Wired to opencast-composer-service-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-videosegmenter-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-videosegmenter-ffmpeg/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-videosegmenter-workflowoperation/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-waveform-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-waveform-ffmpeg/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-waveform-workflowoperation/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-workflow-service-api/15.0.0.SNAPSHOT (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    opencast-workflow-service-impl/15.0.0.SNAPSHOT (Wired to opencast-workflow-service-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-workflow-workflowoperation/15.0.0.SNAPSHOT (Wired to opencast-distribution-service-api/15.0.0.SNAPSHOT which is being refreshed)
    opencast-working-file-repository-service-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-working-file-repository-service-impl/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-workspace-api/15.0.0.SNAPSHOT (Wired to opencast-common/15.0.0.SNAPSHOT which is being refreshed)
    opencast-workspace-impl/15.0.0.SNAPSHOT (Wired to opencast-shared-filesystem-utils/15.0.0.SNAPSHOT which is being refreshed)
    org.apache.aries.jpa.api/2.7.3 (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    org.apache.aries.jpa.blueprint/2.7.3 (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    org.apache.aries.jpa.container/2.7.3 (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    org.apache.aries.jpa.eclipselink.adapter/2.7.3 (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    org.apache.aries.jpa.javax.persistence_2.1/2.7.3 (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    org.apache.aries.jpa.support/2.7.3 (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    org.apache.cxf.cxf-core/3.5.5 (Wired to org.springframework.osgi.core/1.2.1 which is being refreshed)
    org.apache.cxf.cxf-rt-frontend-jaxrs/3.5.5 (Wired to org.springframework.context/3.1.4.RELEASE which is being refreshed)
    org.apache.cxf.cxf-rt-management/3.5.5 (Wired to org.apache.cxf.cxf-core/3.5.5 which is being refreshed)
    org.apache.cxf.cxf-rt-rs-client/3.5.5 (Wired to org.springframework.context/3.1.4.RELEASE which is being refreshed)
    org.apache.cxf.cxf-rt-rs-extension-providers/3.5.5 (Wired to org.apache.cxf.cxf-core/3.5.5 which is being refreshed)
    org.apache.cxf.cxf-rt-rs-extension-search/3.5.5 (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    org.apache.cxf.cxf-rt-rs-json-basic/3.5.5 (Wired to org.apache.cxf.cxf-core/3.5.5 which is being refreshed)
    org.apache.cxf.cxf-rt-rs-service-description/3.5.5 (Wired to org.apache.cxf.cxf-core/3.5.5 which is being refreshed)
    org.apache.cxf.cxf-rt-security/3.5.5 (Wired to org.apache.cxf.cxf-core/3.5.5 which is being refreshed)
    org.apache.cxf.cxf-rt-transports-http/3.5.5 (Wired to org.springframework.context/3.1.4.RELEASE which is being refreshed)
    org.apache.cxf.cxf-rt-transports-http-jetty/3.5.5 (Wired to org.springframework.context/3.1.4.RELEASE which is being refreshed)
    org.apache.cxf.karaf.cxf-karaf-commands/3.5.5 (Wired to org.apache.cxf.cxf-core/3.5.5 which is being refreshed)
    org.apache.servicemix.bundles.aws-java-sdk/1.12.269.1 (Wired to org.springframework.context/3.1.4.RELEASE which is being refreshed)
    org.eclipse.persistence.core/2.7.11.v20220804-52dea2a3c0 (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    org.eclipse.persistence.jpa/2.7.11.v20220804-52dea2a3c0 (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    org.ops4j.pax.transx.pax-transx-tm-geronimo/0.5.3 (Wired to org.springframework.transaction/3.1.4.RELEASE which is being refreshed)
    org.springframework.context/3.1.4.RELEASE (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    org.springframework.context.support/3.1.4.RELEASE (Wired to org.springframework.context/3.1.4.RELEASE which is being refreshed)
    org.springframework.ldap/1.3.1.RELEASE (Wired to org.springframework.context/3.1.4.RELEASE which is being refreshed)
    org.springframework.osgi.core/1.2.1 (Wired to org.springframework.context/3.1.4.RELEASE which is being refreshed)
    org.springframework.osgi.extender/1.2.1 (Wired to org.springframework.context/3.1.4.RELEASE which is being refreshed)
    org.springframework.security.config/3.1.7.RELEASE (Wired to org.springframework.context/3.1.4.RELEASE which is being refreshed)
    org.springframework.security.core/3.1.7.RELEASE (Wired to org.springframework.context/3.1.4.RELEASE which is being refreshed)
    org.springframework.security.ldap/3.1.7.RELEASE (Wired to org.springframework.context/3.1.4.RELEASE which is being refreshed)
    org.springframework.security.web/3.1.7.RELEASE (Wired to org.springframework.context/3.1.4.RELEASE which is being refreshed)
    org.springframework.transaction/3.1.4.RELEASE (Wired to org.springframework.context/3.1.4.RELEASE which is being refreshed)
    org.springframework.web/3.1.4.RELEASE (Wired to org.springframework.context/3.1.4.RELEASE which is being refreshed)
```

If you look closely you see that the only bundle refreshed not just because of transitivity is this:

```

    javax.persistence/2.2.0.v201708071007 (Should be wired to: org.apache.aries.jpa.container/2.7.3 (through [javax.persistence/2.2.0.v201708071007] osgi.wiring.package; filter:="(&(osgi.wiring.package=org.osgi.service.jpa)(version>=1.1.0))"))
```

which then tears down the entirety of Opencast. As said, this is not really an Opencast issue, though. You can reproduce that refresh in a fresh copy of Karaf:

```
karaf@root()> feature:install eclipselink
karaf@root()> feature:install -v -t foo
Adding features: foo/[1.0.0,1.0.0]
No deployment change.
  Bundles to refresh:
    javax.persistence/2.2.0.v201708071007 (Should be wired to: org.apache.aries.jpa.container/2.7.3 (through [javax.persistence/2.2.0.v201708071007] osgi.wiring.package; filter:="(&(osgi.wiring.package=org.osgi.service.jpa)(version>=1.1.0))"))
    org.apache.aries.jpa.api/2.7.3 (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    org.apache.aries.jpa.container/2.7.3 (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    org.apache.aries.jpa.eclipselink.adapter/2.7.3 (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    org.apache.aries.jpa.javax.persistence_2.1/2.7.3 (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    org.apache.aries.jpa.support/2.7.3 (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    org.eclipse.persistence.core/2.7.11.v20220804-52dea2a3c0 (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
    org.eclipse.persistence.jpa/2.7.11.v20220804-52dea2a3c0 (Wired to javax.persistence/2.2.0.v201708071007 which is being refreshed)
```

---

Note you need to configure a feature repo containing an empty feature `foo`.

**Comment by lkiesow** [comment]
This should be fixed by #5153.
Please re-open if this still persists.

**Comment by JulianKniephoff** [comment]
The intention of this issue was to track the upstream bug (once I get to reporting that ...) and to remind me (and/or others) to remove the workaround (i.e. #5153) once it's fixed. Also the elusive features that depend on the refresh are still broken now, of course.

So on these grounds I would personally reopen this. What do you think @lkiesow?
